### ISSUE TITLE
USWDS - Memorable date: Input formatting requirements are unclear to users of screen readers.

### ISSUE THREAD

**Issue by jaclinec** [uswds/uswds#5902]
# Summary
The memorable date hint messaging announced to screen readers is overly described and unclear:

1. It announces the same hint for each field ("For example, January 19 2000" is announced when the user lands on the month field, the day field, and the year field).
2. It doesn’t describe the digits/format needed for inputs, so users are left guessing what to type. 

# Observations
2 participants (using JAWS and NVDA) experienced friction with understanding the formatting required for the memorable date inputs because the hint announced was not explicit on what is expected for each input and digit requirements. See [video clips for memorable date version 1](https://drive.google.com/file/d/19bhqXwg-ILD_QqDYiAfxp6D27jElSOdQ/view?usp=sharing). :lock:

The hint message announced in another version of memorable date that we tested performed much better since the hints were specific to the formatting expectations for each input. See [video clips for memorable date version 2](https://drive.google.com/file/d/1InerMwidDiddYnAKPHLPkUmXg1HdTkU2/view?usp=sharing) :lock:

# Affected user groups

- Screen reader users

# Research method
Usability testing with 8 participants with disabilities: 
- 5 visually impaired (3 blind, 2 low vision/legally blind (1 with nystagmus))
- 1 motor impairment
- 1 ADHD
- 1 anxiety and depression

Screen reader software participants used: JAWS, NVDA, and VoiceOver

[Link to public findings report forthcoming...here is the internal [Zebra findings report](https://docs.google.com/document/d/1vbGSh4UNZiseyzdGXa1pJwU_3M4y7HaU3QPkXOa7gsI/edit?usp=sharing)] :lock:

# Next steps

Change the hint message and announcement to screen readers to be relevant to each input and clear on digit requirements. 

Move forward with the following changes:

<img width="650" alt="Image" src="https://github.com/user-attachments/assets/edaef92b-deda-4daf-9caf-6b282a937314" />

Hint text should read: 

> Select a month from the dropdown. Enter two digits for the day and four digits for the year.
> For example: January 01 2000"

Screen reader announcement should say:

Month input:

> Select a month from the dropdown

Day input:

> Enter two digits for the day

Year input:

> Enter four digits for the year

**Comment by annepetersen** [comment]
So what we need next here seems to be content, supported by accessibility and UX. Specifically: 

- [ ] the phrasing of the hint for the month field, the day field, and the year field
- [ ] how we should describe the digits/format needed for inputs

The three of you will need to get together to figure out these changes before we get this into a sprint. (Which, process-wise, will always be necessary when there are text changes like this, so @jaclinec, that's a next step you can always initiate to keep these issues moving)

**Comment by alex-hull** [comment]
Adding this as a point of discussion for our next UXR A11y sync. @jaclinec @amycole501 @cfpollack

**Comment by jaclinec** [comment]
Removing January user testing milestone since we don't plan to test memorable date in the January round of usability testing. Setting for February release and we will test it in the following round of testing. If we see any issues in testing, we can correct them at a later date.

**Comment by finekatie** [comment]
Hi @jaclinec , should I put this issue on hold then?

**Comment by jaclinec** [comment]
To clarify the next steps for this issue, we need to determine:

1. **The hint text that displays above the memorable date inputs**. Right now it says "For example, January 19 2000" but should be changed to something like "Select a month from the dropdown. Enter two digits for the day and four digits for the year." or similar.
2. **The hint announcement to screen readers** needs to change to be more relevant depending on the input the user lands on. Right now, it announces "For example, January 19 2000" no matter what input they focus on. Instead, it should state "Type the month name or select from the dropdown" when they focus on the month combobox, "Day, enter two digits" when they focus on the day, and "year, enter four digits" when they focus on the year.

Let's brainstorm ideas and bring it to a CS UX a11y sync.

**Comment by jaclinec** [comment]
Removed @cfpollack , @alex-hull and @finekatie as Katie already reviewed and approved my suggestions above and the next step is taking the suggestions to an upcoming CS UX a11y sync to get the team's and Feds' opinions. Adding it to the meeting agenda.

**Comment by jaclinec** [comment]
After discussing in CS UX sync today, we decided to move forward with the following changes:

<img width="650" alt="Image" src="https://github.com/user-attachments/assets/edaef92b-deda-4daf-9caf-6b282a937314" />


Hint text should read: 

> Select a month from the dropdown. Enter two digits for the day and four digits for the year.
> For example: January 01 2000"

Screen reader announcement should say:

Month input:

> Select a month from the dropdown

Day input:

> Enter two digits for the day

Year input:

> Enter four digits for the year

**Comment by jaclinec** [comment]
Once these changes are made, we can test it in another round of usability testing. I don't think we need to test it before implementing (we already tested a version of the improved language in the Zebra batch testing) - we can launch and then validate later.

Passing this issue over to development. Please reassign as needed and possibly add to next sprint if bandwidth allows.

CC: @annepetersen @thisisdano @amyleadem @mejiaj @mahoneycm @dct2023

**Comment by dct2023** [comment]
@jaclinec Thank you for the note! Could you update the issue description to add the new details and next steps?

**Comment by jeana-adhoc** [comment]
VA Design System has integrated this change into our component. You can see the [example here in our storybook.](https://design.va.gov/storybook/?path=/story/uswds-va-memorable-date--with-month-select). We left off the `For example: January 01 2000` text so that the text that is announced matches the visible text. There was also confusion if we should add a comma in the copy so it would read, `For example: January 01, 2000" but Accessibility specialists were concern this could instruct users to input a comma in the field, and that would invalidate the information.

**Comment by chandracarney** [comment]
Thank you for all the work thus far, we're planning to start on this.
